Juriba.Dashworks/Public/New-DwDashboardHalfDonutWidget.ps1

```diff
@@ -0,0 +1,66 @@
+Function New-DwDashboardHalfDonutWidget {
+
+    [CmdletBinding(SupportsShouldProcess)]
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$Instance,
+        [Parameter(Mandatory=$true)]
+        [string]$APIKey,
+        [Parameter(Mandatory=$true)]
+        [int]$DashboardId,
+        [Parameter(Mandatory=$true)]
+        [int]$SectionId,
+        [Parameter(Mandatory=$true)]
+        [string]$Title,
+        [Parameter(Mandatory=$true)]
+        [int]$ListId,
+        [Parameter(Mandatory=$true)]
+        [string]$SplitBy,
+        [Parameter(Mandatory=$false)]
+        [bool]$OrderByDescending = $false,
+        [Parameter(Mandatory=$false)]
+        [string]$OrderByField = $null
+    )
+
+    $request = @{
+        Uri         = ($Instance + "/apiv1/dashboard/" + $DashboardId + "/section/" + $SectionId + "/widget")
+        Method      = "Post"
+        Body        = @{
+            "widgetType"                = "HalfDonut"
+            "displayOrder"              = 0
+            "title"                     = $Title
+            "listId"                    = $ListId
+            "listObjectType"            = "devices"
+            "listObjectTypeId"          = 2
+            "colourSchemeId"            = 1
+            "widgetValuesColours"       = @()
+            "colourTemplateId"          = 1
+            "aggregateFunctionId"       = 1
+            "maximumValues"             = 10
+            "drilldownEnabled"          = $true
+            "maxRows"                   = 0
+            "maxColumns"                = 0
+            "cardTypeIsAggregate"       = $false
+            "splitBy"                   = $SplitBy
+            "aggregateBy"               = ""
+            "orderByDescending"         = if ($OrderByDescending) { $true }else { $false }
```

Comment:
The definition of OrderByDescending is:
 
 [bool]$OrderByDescending = $false
 
 thus I don't see how you'd need more than:
 
 "orderByDescending"  = $OrderByDescending
 
 Given that the variable has a default value of false. It does the same thing.

Reply:
Good point, changed.